src/AnalyticsContext.tsx

```diff
@@ -35,18 +55,47 @@ const AnalyticsProvider: FunctionComponent = ({ children }) => {
     checkAnalyticsConsent()
   }, [userConsentedToAnalytics])
 
+  useEffect(() => {
+    const supportAnalyticsTracking =
+      healthAuthoritySupportsAnalytics && userConsentedToAnalytics
+
+    const initializeAnalyticsTracking = () => {
```

Comment:
say a user updates their consent multiple times would we initialize this multiple times?   Should this be behind a subscription?